src/copy-paste.js

```diff
@@ -0,0 +1,347 @@
+            // HACK fix intermittent multiple-paste on Chrome (timing related?)
+            hiddenTextarea.val("");
+        }
+        return value;
+    }
+
+    function onCut(opts) {
+        var $focus = $(':focus');
+        if ($focus.is('.jstree-anchor')) {
+            var text = opts.cut();
+            if (text) {
+                focusTextarea($focus, text);
+                setTimeout(function () {
+                    unfocusTextarea($focus);
+                }, 10);
+            }
+        }
+    }
+
+    function onCopy(opts) {
+        var $focus = $(':focus');
+        if ($focus.is('.jstree-anchor')) {
+            var text = opts.copy();
+            if (text) {
+                focusTextarea($focus, text);
+                setTimeout(function () {
+                    unfocusTextarea($focus);
+                }, 10);
+            }
+        }
+    }
+
+    function onPaste(opts) {
+        var $focus = $(':focus');
+        if ($focus.length === 0 || $focus.parents('.fd-tree').length ||
+                                   $focus.is(hiddenTextarea)) {
+            focusTextarea($focus);
+            setTimeout(function () {
+                var pasteValue = unfocusTextarea($focus, true);
+                // on chrome this gets called twice,
+                // the first time with a blank value
+                if (pasteValue) {
+                    var errors = opts.paste(pasteValue);
+                    if (errors.length) {
+                        vellum._resetMessages([{
+                            level: "parse-warning",
+                            message: errors,
+                        }]);
+                    }
+                }
+            }, 0);
+        }
+    }
+
+    // matches strings that could be JSON; see http://json.org/
+    var JSON_STRING = /^(null|true|false|\[[^]*\]|\{[^]*\}|"[^]*"|-?\d+(\.\d+)?([Ee][+-]?\d+)?)$/;
+
+    /**
+     * Convert value to string
+     *
+     * Does nothing to strings that do not look like JSON. The idea
+     * behind this (along with valuify) is to make the output as human
+     * readable as possible while maintaining lossless de/serialization.
+     */
+    function stringify(value) {
+        if (_.isString(value) && !JSON_STRING.test(value)) {
+            return value;
+        }
+        return JSON.stringify(value);
+    }
+
+    /**
+     * Convert string to value
+     *
+     * Strings that look like JSON will be parsed as JSON, otherwise the
+     * string value is passed through unchanged.
+     */
+    function valuify(string) {
+        if (JSON_STRING.test(string)) {
+            return JSON.parse(string);
+        }
+        return string;
+    }
+
+    function headerKey(item) {
+        var rank = "1",
+            itext = /^(.*)Itext(?::([^-]+)-(.*))?$/.exec(item);
+        if (itext) {
+            rank = "0"; // itext before other fields
+            // label before all other itext types
+            if (itext[1] === "label") { itext[1] = "!"; }
```

Comment:
small note, sometimes i like to label my indices for more readability:
```
var TYPE = 1,
   LANG = 2,
   FORM =3;
itext[TYPE] == 'something'
```  
totally your call